frontend/src/pages/EditStudent.vue

```diff
@@ -0,0 +1,344 @@
+                    </b-input>
+                </b-field>
+                <b-field grouped>
+
+                  <b-field label="Native Language" class="half-width">
+                     <b-autocomplete v-model="studentData.NativeLanguageString"
+                        field= "NativeLanguage"
+                        ref="languageComplete"
+                        :data="languages"
+                        placeholder="e.g. Bengali" 
+                        @typing="filteredLanguageDataArray"
+                        @select="option => selected = option">
+                        <template slot="header">
+                            <a @click="showAddLanguage">
+                                <span> Add new... </span>
+                            </a> 
+                        </template>
+                    </b-autocomplete>
+                  </b-field>
+
+          
+                  <b-field label="English Proficiency" class="half-width">
+                      <b-select v-model="studentData.EnglishProficiency" placeholder="Select one" expanded>
+                        <option value = "No">No (Unable to understand at all)</option>
+                        <option value = "Little">Little (Able to understand simple words)</option>
+                        <option value = "Simple">Simple (Able to speak full sentences)</option>
+                        <option value = "Intermediate">Intermediate (Able to understand simple words)</option>
+                      </b-select>
+                  </b-field>  
+                </b-field>
+        
+                <b-field label="Notes" class="half-width">
+                    <b-input v-model="studentData.Notes" maxlength="200" type="textarea" placeholder="Optional"></b-input>
+                </b-field>
+
+            </section>
+            <b-button class="dark-blue" input type="submit" expanded @click="saveStudent">Save Student</b-button>
+          </form>
+        </div>
+      </div>
+    </div>  
+
+  </Page>    
+</template>
+
+
+<script>
+
+import Page from '../components/Page.vue'
+export default {
+  name: 'EditStudent',
+    components: {
+    Page,
+  },
+
+  data() {
+      return {
+          name: '',
+          PhoneNumber: '',
+          source:'',
+          EnglishProficiency: '',
+          file: null,
+          Notes: '',
+          selected: {
+            NativeLanguage: ''
+          },
+          studentData: {},
+          languages: [],
+          API_nativeLanguage: []
+
+      }
+  },
+  computed: {
+    // If NativeLanguage is changed, we assign it this value 
+    nativeLanguage(){
+      return this.selected ? this.selected.NativeLanguage: ''
+    }
+  },
+  watch: {
+    file (val){
+      this.uploadFile() 
+    }
+  },
+
+ async mounted() {
+    // Call API for Native Languages
+      await fetch("/api/nativeLanguages")
+        .then(response => response.json())
+        .then(result => this.API_nativeLanguage = result)
```

Comment:
Better not to mix `await` and `.then` syntax.

```suggestion
  const nativeLanguageResponse = await fetch("api/nativeLanguages")
  const nativeLanguageData = await nativeLanguageResponse.json()
  this.API_nativeLanguage = nativeLanguageData
```

Both the `fetch` function and `.json()` method return promises, that's why it is necessary to `await` them.